cmd/controller/apps_reconciler.go

```diff
@@ -47,6 +48,7 @@ func (r *AppsReconciler) Reconcile(ctx context.Context, request reconcile.Reques
 	}
 
 	crdApp := r.appFactory.NewCRDApp(existingApp, log)
+	metrics.InitMetrics(existingApp.Name, existingApp.Namespace)
```

Comment:
i was looking at bit more around what code is required to run this metrics service. i do think we should inject an object that holds appropriate fields. seems like having it be a singleton is not a great thing.

Reply:
would you be interested in tweaking it to make metrics a separate objects, or we could potentially change that ourselves?